terror_error.go

```diff
@@ -291,3 +296,35 @@ func (e *Error) UnmarshalJSON(data []byte) error {
 	}
 	return nil
 }
+
+func (e *Error) WarpCauseError(err error) *Error {
+	e.cause = err
+	return e
+}
+
+func (e *Error) Cause() error {
+	return e.cause
```

Comment:
The cause error may be another wrapper for the real root cause. Do we need to handle this situation?